ngx-tools/coercion/src/number/number-property.ts

```diff
@@ -19,7 +19,7 @@ export function coerceNumberProperty(value: any, fallbackValue = 0) {
  * Whether the provided value is considered a number.
  * @docs-private
  */
-export function _isNumberValue(value: any): boolean {
+export function _isNumberValue(value: any): value is number {
```

Comment:
Same note as date coercion.. just need to add coercion to the function overloads above:

```typescript
export function coerceNumberProperty(value: any): value is number;
export function coerceNumberProperty<D>(value: any, fallback: D): value is number | D;
```